staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache_test.go

```diff
@@ -333,10 +371,38 @@ func TestWaitUntilFreshAndList(t *testing.T) {
 	if len(list) != 2 {
 		t.Errorf("unexpected list returned: %#v", list)
 	}
+
+	// list with spec.nodeName index.
+	matchValues = []storage.MatchValue{
+		{IndexName: "l:not-exist-label", Value: "whatever"},
+		{IndexName: "f:spec.nodeName", Value: "node2"},
+	}
+	list, resourceVersion, err = store.WaitUntilFreshAndList(5, matchValues, nil)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if resourceVersion != 5 {
+		t.Errorf("unexpected resourceVersion: %v, expected: 5", resourceVersion)
+	}
+	if len(list) != 1 {
```

Comment:
Wait - why is that 1?
Given that in WaitUntilFreshAndList you go over `matchValue` in the order it is set, the first one you check is the index: "l:not-exist-label=whatever". For this there shouldn't be any pods matching this selector.
So isn't that 0?

Reply:
the cacher has no l:not-exist-label index. so the first one will return error, it will try the second matchValue. it will return 1

Reply:
OK - never mind then.